CHANGELOG.md

```diff
@@ -36,6 +36,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [Autotagging] Fix delete rule event consumption in InMemoryRuleProcessingService ([#18628](https://github.com/opensearch-project/OpenSearch/pull/18628))
 - Cannot communicate with HTTP/2 when reactor-netty is enabled ([#18599](https://github.com/opensearch-project/OpenSearch/pull/18599))
 - Fix the visit of sub queries for HasParentQuery and HasChildQuery ([#18621](https://github.com/opensearch-project/OpenSearch/pull/18621))
+- Honor COMPLEMENT for Regexp queries ([#18640](https://github.com/opensearch-project/OpenSearch/pull/18640))
```

Comment:
Can you please mention in the changelog that this fixes a backward compatibility regression introduced in OpenSearch 3.0, but will be removed in OpenSearch 4.0?

Right now, someone could get the impression that this is a permanent fix.